### PR TITLE
Clean up the Makefile

### PR DESCRIPTION
**Pull request by poncovka**
Reorganize the code and remove some useless targets.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -17,72 +17,57 @@
...
 	@echo "*** Running unittests with $(COVERAGE) ***"
 	PYTHONPATH=. $(COVERAGE) run --branch -m unittest discover -v -s tests/
 	$(COVERAGE) report -m --include="dasbus/*" | tee coverage-report.log
 
-check:
-	@echo "*** Running pylint ***"
-	$(PYTHON) -m pylint dasbus/ tests/
+.PHONY: changelog
+changelog:
+	@git log --pretty="format:- %s (%ae)" $(TAG).. | sed -e 's/@.*)/)/' | grep -v "Merge pull request"
```
Comment (jkonecny12) [inline Makefile]:
```suggestion
	@git log --no-merges --pretty="format:- %s (%ae)" $(TAG).. | sed -e 's/@.*)/)/'
```

Seems like a better solution to me.

File: Makefile [hunk 2]
```diff
@@ -17,72 +17,57 @@
...
-	sed -i "s/Version:   $(VERSION)/Version:   $$NEWVERSION/" python-${PKGNAME}.spec ; \
-	sed -i "s/version='$(VERSION)'/version='$$NEWVERSION'/" setup.py
+.PHONY: archive
+archive:
+	$(PYTHON) setup.py sdist bdist_wheel
+	@echo "The archive is in dist/$(PKGNAME)-$(VERSION).tar.gz"
 
-.PHONY: clean install tag archive local
+.PHONY: upload
+upload:
```
Comment (jkonecny12) [inline Makefile]: I would recommend to change this to ``upload-pypi:``.

File: Makefile [hunk 3]
```diff
@@ -17,72 +17,57 @@
...
-	@rm -rf $(PKGNAME)-$(VERSION).tar.gz
-	@rm -rf /tmp/$(PKGNAME)-$(VERSION) /tmp/$(PKGNAME)
-	@dir=$$PWD; cp -a $$dir /tmp/$(PKGNAME)-$(VERSION)
-	@cd /tmp/$(PKGNAME)-$(VERSION) ; $(PYTHON) setup.py -q sdist
-	@cp /tmp/$(PKGNAME)-$(VERSION)/dist/$(PKGNAME)-$(VERSION).tar.gz .
-	@rm -rf /tmp/$(PKGNAME)-$(VERSION)
-	@echo "The archive is in $(PKGNAME)-$(VERSION).tar.gz"
+.PHONY: push
+push:
+	git push --follow-tags
```
Comment (jkonecny12) [inline Makefile]: I would be little worried about making push by Makefile. But this is your call.

Comment (poncovka) [reply Makefile]: Fixed.

File: Makefile (hunk 2 above)
Comment (poncovka) [reply Makefile]: I will keep it as it is for now.

**Pull request merged**